up_client_socket/rust/u_transport_socket.rs

```diff
@@ -0,0 +1,395 @@
+            TcpStreamSync::connect(DISPATCHER_ADDR).expect("issue in connecting  sync socket");
+
+        UtransportSocket {
+            socket_sync,
+            listner_map: Arc::new(Mutex::new(HashMap::new())),
+        }
+    }
+}
+
+impl UtransportExt for UtransportSocket {
+     fn socket_init(&mut self) {
+        loop {
+            // Receive data from the socket
+            let mut buffer: [u8; BYTES_MSG_LENGTH] = [0; BYTES_MSG_LENGTH];
+
+            let bytes_read = match self.read_socket(&mut buffer) {
+                Ok(bytes_read) => bytes_read,
+                Err(e) => {
+                    dbg!("Socket error: {}", e);
+                    break;
+                }
+            };
+
+            // Check if no data is received
+            if bytes_read == 0 {
+                continue;
+            }
+
+            let umessage = UMessage::parse_from_bytes(&buffer[..bytes_read]).expect("Failed to parse message");
+
+            match umessage.attributes.type_.enum_value() {
+                Ok(mt) => match mt {
+                    UMessageType::UMESSAGE_TYPE_PUBLISH => {
+                        println!("calling handle publish...");
+                        self._handle_publish_message(umessage);
+                        ()
+                    }
+                    UMessageType::UMESSAGE_TYPE_NOTIFICATION => todo!(),
+                    UMessageType::UMESSAGE_TYPE_UNSPECIFIED => (),
+                    UMessageType::UMESSAGE_TYPE_RESPONSE => (),
+                    UMessageType::UMESSAGE_TYPE_REQUEST => {
+                        self._handle_request_message(umessage);
+                        ()
+                    }
+                },
+                Err(_) => (),
+            }
+        }
+    }
+
+    fn read_socket(&self, buffer: &mut [u8]) -> io::Result<usize> {
+        let mut socket = &self.socket_sync;
+
+        socket.read(buffer)
+    }
+
+     fn _handle_publish_message(&mut self, umsg: UMessage) {
+        dbg!("HANDLING PUB MESG");
+        // Create a new Tokio runtime
+        let rt = Runtime::new().unwrap();
+   
+        dbg!(&umsg.attributes.source.to_string());
+    
+        if let Some(listner_array) = self
+            .listner_map
+            .lock()
+            .unwrap()
+            .get(&umsg.attributes.source.to_string())
+        {
+            for listner_ref in listner_array {
+                rt.block_on(async {
+                    let _ = listner_ref.on_receive(umsg.clone()).await;
+                });
+            }
+        }
+    }
+
+     fn _handle_request_message(&mut self, umsg: UMessage) {
+
+        let rt_ = Runtime::new().unwrap();
+        if let Some(listner_array) = self
+            .listner_map
+            .lock()
+            .unwrap()
+            .get(&umsg.attributes.sink.to_string())
+        {
+            for listner_ref in listner_array {
+                rt_.block_on(async {
+                let _ = listner_ref.on_receive(umsg.clone()).await;
```

Comment:
`on_receive()` doesn't return anything, so we can:
```suggestion
                    listner_ref.on_receive(umsg.clone()).await;
```